src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -485,4 +558,150 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    /**
+     * sorts posts by following criteria
+     * 1. criterion: displayPriority is PINNED -> pinned posts come first
+     * 2. criterion: displayPriority is ARCHIVED  -> archived posts come last
+     * -- in between pinned and archived posts --
+     * 3. criterion: currently selected criterion in combination with currently selected order
+     *
+     * @param postA             post 1 to be compared
+     * @param postB             post 2 to be compared
+     * @param postSortCriterion criterion to sort posts (CREATION_DATE, #VOTES,#ANSWERS)
+     * @param sortingOrder      direction of sorting (ASC, DESC)
+     * @return number indicating the order of two elements
+     */
+    public static int postComparator(Post postA, Post postB, PostSortCriterion postSortCriterion, SortingOrder sortingOrder) {
+
+        // sort by priority
+        Integer order = sortByPriority(postA, postB);
+        if (order != null) {
+            return order;
+        }
+
+        // sort by votes via voteEmojiCount
+        if (postSortCriterion == PostSortCriterion.VOTES) {
+            order = sortByVotes(postA, postB, sortingOrder);
+            if (order != null) {
+                return order;
+            }
+        }
+
+        // sort by creation date
+        if (postSortCriterion == PostSortCriterion.CREATION_DATE) {
+            order = sortByCreationDate(postA, postB, sortingOrder);
+            if (order != null) {
+                return order;
+            }
+        }
+
+        // sort by answer count
+        if (postSortCriterion == PostSortCriterion.ANSWER_COUNT) {
+            order = sortByAnswerCount(postA, postB, sortingOrder);
+            if (order != null) {
+                return order;
+            }
+        }
+        return 0;
+    }
+
+    @Nullable
+    private static Integer sortByPriority(Post postA, Post postB) {
```

Comment:
I updated the method's name. If you go one method up, you may see that 0 instead of null is returned after all different comparisons if the posts are equal in priority.

Reply:
Yes, but is there some reason why only the full comparison method should adhere to the `Comparable` convention? Wouldn’t it make sense that all helper methods also use this convention of returning `0` if the elements are considered to be equal according to this specific comparison?

Reply:
Yes, I agree with your statement. It makes sense to follow the convention for the sub-methods as well. I updated the code.